### PR TITLE
[MCH-281-140] Feat: solidly swapper

### PR DESCRIPTION
**Pull request by gotzenx**

### CODE REVIEW

File: packages/yswaps/scripts/libraries/solvers/multicall/BooSolidSeller.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    console.log('[BooSolidSeller] Getting solid => wftm trade information');
+    const path = [
+      {
+        from: solid.address,
+        to: wftm.address,
+        stable: false,
+      },
+    ];
+    let calculatedWftmAmount = (await solidlyRouter.getAmountsOut(solidBalance, path))[1];
+    calculatedWftmAmount = calculatedWftmAmount.sub(calculatedWftmAmount.div(100).mul(1).div(2));
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/libraries/solvers/multicall/BooSolidSeller.ts:62]: I did this code, but yikes lol

**Review by b0dhidharma — APPROVED** [review]
gw mah man!

File: packages/yswaps/scripts/libraries/solvers/SolidlySolver.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      IERC20Metadata__factory.connect(tokenOutAddress, tradeFactory.signer),
+    ]);
+
+    const [inSymbol, inDecimals, outSymbol, outDecimals, amount] = await Promise.all([
+      tokenIn.symbol(),
+      tokenIn.decimals(),
+      tokenOut.symbol(),
+      tokenOut.decimals(),
+      tokenIn.balanceOf(strategy),
+    ]);
```
[the remark this answers is not in the document]
Comment (gotzenx) [reply packages/yswaps/scripts/libraries/solvers/SolidlySolver.ts:40]: fixed

File: packages/yswaps/deploy/fantom-swappers/solidly.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  //     args: [governor, tradeFactory.address, WETH, WFTM, SPIRITSWAP_FACTORY, SPIRITSWAP_ROUTER],
+  //     log: true,
+  //   });
+
+  //   if (await shouldVerifyContract(syncDeploy)) {
+  //     await hre.run('verify:verify', {
+  //       address: syncDeploy.address,
+  //       constructorArguments: [governor, tradeFactory.address, WETH, WFTM, SPIRITSWAP_FACTORY, SPIRITSWAP_ROUTER],
+  //     });
+  //   }
```
[the remark this answers is not in the document]
Comment (gotzenx) [reply packages/yswaps/deploy/fantom-swappers/solidly.ts:41]: I created the linear ticket for the Sync swapper. Should we keep the comment?

File: packages/yswaps/test/integration/swappers/async/solidly.spec.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+  let USDC: IERC20;
+  let WFTM: IERC20;
+
+  let snapshotId: string;
+
+  let solidlyResponse: SwapResponse;
+
+  when('on fantom', () => {
+    // const FORK_BLOCK_NUMBER = forkBlockNumber['mainnet-swappers'];
```
[the remark this answers is not in the document]
Comment (gotzenx) [reply packages/yswaps/test/integration/swappers/async/solidly.spec.ts]: updated the block number for fantom fork and enabled this line again

**Pull request merged**